PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/table/AppSettingsTableUtils.java

```diff
@@ -0,0 +1,182 @@
+
+        final AnActionButton exportButton = new AnActionButton(EXPORT_LOCAL_SETTINGS_TITLE, AllIcons.General.ExportSettings) {
+            @Override
+            public void actionPerformed(AnActionEvent anActionEvent) {
+                try {
+                    final File file = DefaultLoader.getUIHelper().showFileSaver(EXPORT_APP_SETTINGS, LOCAL_SETTINGS_JSON);
+                    AppSettingsTableUtils.exportLocalSettingsJsonFile(file, appSettingsTable.getAppSettings());
+                } catch (IOException e) {
+                    MvpUIHelperFactory.getInstance().getMvpUIHelper().showException(EXPORT_APP_SETTINGS_FAILED, e);
+                }
+            }
+        };
+
+        appSettingsTable.getSelectionModel().addListSelectionListener(new ListSelectionListener() {
+            @Override
+            public void valueChanged(ListSelectionEvent listSelectionEvent) {
+                final String prompt = AzureFunctionsConstants.getAppSettingHint(appSettingsTable.getSelectedKey());
+                promptPanel.setText(prompt);
+            }
+        });
+
+        appSettingsTable.addFocusListener(new FocusListener() {
+            @Override
+            public void focusGained(FocusEvent focusEvent) {
+                final String prompt = AzureFunctionsConstants.getAppSettingHint(appSettingsTable.getSelectedKey());
+                promptPanel.setText(prompt);
+            }
+
+            @Override
+            public void focusLost(FocusEvent focusEvent) {
+                promptPanel.setText("");
+            }
+        });
+
+        final ToolbarDecorator tableToolbarDecorator = ToolbarDecorator.createDecorator(appSettingsTable)
+                .addExtraActions(btnAdd, btnRemove, importButton, exportButton).setToolbarPosition(ActionToolbarPosition.RIGHT);
+        final JPanel tablePanel = tableToolbarDecorator.createPanel();
+        final GridConstraints tableConstraint = new GridConstraints(0, 0, 1, 1, 0, GridConstraints.FILL_BOTH, 7, 7, null, null, null);
+        result.add(tablePanel, tableConstraint);
+        result.setMinimumSize(new Dimension(-1, 100));
+        return result;
+    }
+
+    public static Map<String, String> getAppSettingsFromLocalSettingsJson(File target) {
+        final Map<String, String> result = new HashMap<>();
+        try {
+            final JsonObject jsonObject = readLocalSettingsJsonFile(target);
+            final JsonObject valueObject = jsonObject.getAsJsonObject(LOCAL_SETTINGS_VALUES);
+            valueObject.entrySet().forEach(entry -> result.put(entry.getKey(), entry.getValue().getAsString()));
+            return result;
+        } catch (JsonParseException e) {
+            return new HashMap<>();
+        }
+    }
+
+    public static void exportLocalSettingsJsonFile(File target, Map<String, String> appSettings) throws IOException {
+        if (target == null) {
+            return;
+        }
+        if (!target.exists()) {
+            target.createNewFile();
```

Comment:
shall we check parent folder existence?

Reply:
Path `xxx` does not exist, please check the path and try again.

Reply:
Added the check and will create parent folder if not exists